R/analytical_functions.R

```diff
@@ -269,9 +277,53 @@ life.table.df <- function(necdf, agecor = TRUE, agecorfac = c()) {
     )
   }
 
+
   # dx: proportion of deaths within x.
   necdf['dx'] <- necdf['Dx'] / sum(necdf['Dx']) * 100
 
+
+  # in case of spline-interpolation, dx-values will be replaced with interpolated once
+
+  if (length(option_spline) > 0) {
+
+  a_cumsum <- necdf['a'] %>% cumsum
+
+  # add index number to resulting data.frame
+  a_cumsum$id <- 1:nrow(a_cumsum)
+
+  repeat_number <- floor((limit - 20) / option_spline)
+
+  a_cumsum_select <- a_cumsum$id[which(a_cumsum$a <= 20)]
+
+  for (t in 1 : repeat_number) {
+    a_cumsum_select <- c(a_cumsum_select, (a_cumsum$id[which(a_cumsum$a == (20 + (t * option_spline)))]))
+  }
+
+  if (((limit - 20) / option_spline) - floor((limit - 20) / option_spline) > 0) {
+    a_cumsum_select <- c(a_cumsum_select, a_cumsum$id[nrow(necdf)])
+  }
+
+  dx_cumsum <- c(cumsum(necdf[, 'dx']))[1:nrow(necdf)]
+  y_spline <- dx_cumsum[a_cumsum_select]
+  x_spline <- a_cumsum$a[a_cumsum_select]
+
+  # interpolating the values with a monotonic cubic spline
+  dem <- cm.spline(x_spline, y_spline, n = (limit/5 + 1), xmin = 0, xmax = limit)
```

Comment:
could be replaced by:
dem <- stats::spline(x_spline, y_spline, n = (limit/5 + 1), xmin = 0, xmax = limit, method = "hyman")